wordpress-mobile/WordPress-Android
# Issue/13329 backup progress

Parent #13329 

This PR adds the Progress view and includes:

- Added recyclerView to `backup_download_progress_fragment.xml`
- Updates to `BackupDownloadProgressFragment` and `BackupDownloadProgressViewModel` to show content for the view.
- New `Common.ViewTypes` and ViewHolders for progress and additional information lines items to support progress view
- New `BackupDownloadProgressStateListItemBuilder` to build the list items for progress
- Introduces the Progress indicator

**Notes**:
- Pay no attention to styling
- Error states are handled as snackbars
- The Click Me button is not hooked up, just shows a snackbar
- There is a delay in the status query which I am still playing with. I see mostly large jumps with 10000L delay: 0 -> 66 -> 88 -> complete. Would love suggestions.

**Merge Instructions**
- Make sure PR #13702 and #13711 have been merged
- Remove Not Ready for Merge label
- Merge as normal

**To Test**
- Launch the app with BackupDownloadConfig flag on
- Navigate to ActivityLog
- Tap the More menu
- Tap Download Backup menu item
- Tap the Create downloadable file button
- Note that the progress view is shown and the progress increments

PR submission checklist:
